app/controllers/user_controller.php

```diff
@@ -0,0 +1,74 @@
+<?php
+class UserController extends AppController
+{
+    const ERROR_VAR                 = 'error';
+    const ERROR_MESSAGE_USERPASS    = 'Incorrect username or password';
+
+    public function create()
+    {
+        authRedirect();
+
+        $user = new User;
+        $page = Param::get('page_next', 'create');
+
+        switch ($page) {
+            case 'create_end':
+                $user->username = trim(Param::get('username'));
+                $user->email    = trim(Param::get('email'));
+                $user->password = Param::get('password');
+
+                try {
+                    $user->create();
+                } catch(ValidationException $e) {
+                    $page = 'create';
+                }
+                break;
+            case 'create':
+                break;
+            default:
+                throw new NotFoundException("{$page} is not found.");
+                break;
+        }
+
+        $title = 'Create User';
+        $this->set(get_defined_vars());
+        $this->render($page);
+    }
+
+    public function authenticate()
+    {
+        authRedirect();
+
+        $page = Param::get('page_next', 'auth');
+
+        switch($page) {
+            case 'auth_end':
+                $username = Param::get('username');
+                $password = Param::get('password');
+
+                if(User::authenticate($username, $password)) {
+                    redirect('/');
+                } else {
+                    $this->set(array(self::ERROR_VAR => self::ERROR_MESSAGE_USERPASS));
+                    $page = 'auth';
+                }
+                break;
+            case 'auth':
+                break;
+            default:
+                throw new NotFoundException("{$page} not found.");
+        }
+
+        $title = 'Log in';
+        $this->set(get_defined_vars());
+        $this->render($page);
+    }
+
+    public function logout()
+    {
+        if(User::getAuthenticated()) {
+            session_destroy();
+        }
+        redirect('/');
```

Comment:
maybe it's better to input the proper url here so that it's easier to know where will this be redirected po :bow: